packages/@aws-cdk/aws-secretsmanager/lib/rotation-schedule.ts

```diff
@@ -39,12 +49,364 @@ export class RotationSchedule extends Resource {
   constructor(scope: Construct, id: string, props: RotationScheduleProps) {
     super(scope, id);
 
+    if (!props.rotationLambda && !props.hostedRotation) {
+      throw new Error('Either `rotationLambda` or `hostedRotation` must be specified.');
+    }
+
     new CfnRotationSchedule(this, 'Resource', {
       secretId: props.secret.secretArn,
-      rotationLambdaArn: props.rotationLambda.functionArn,
+      rotationLambdaArn: props.rotationLambda?.functionArn,
+      hostedRotationLambda: props.hostedRotation?.bind(props.secret, this),
       rotationRules: {
         automaticallyAfterDays: props.automaticallyAfter && props.automaticallyAfter.toDays() || 30,
       },
     });
+
+    // Prevent secrets deletions when rotation is in place
+    props.secret.denyAccountRootDelete();
   }
 }
+
+/**
+ * Single user hosted rotation options
+ */
+export interface SingleUserHostedRotationOptions {
+  /**
+   * A name for the Lambda created to rotate the secret
+   *
+   * @default - a CloudFormation generated name
+   */
+  readonly functionName?: string;
+
+  /**
+   * A list of security to groups for the Lambda created to rotate the secret
+   *
+   * @default - a new security group is created
+   */
+  readonly securityGroups?: ec2.ISecurityGroup[];
+
+  /**
+   * The VPC where the Lambda rotation function will run.
+   *
+   * @default - the Lambda is not deployed in a VPC
+   */
+  readonly vpc?: ec2.IVpc;
+
+  /**
+   * The type of subnets in the VPC where the Lambda rotation function will run.
+   *
+   * @default - the Vpc default strategy if not specified.
+   */
+  readonly vpcSubnets?: ec2.SubnetSelection;
+}
+
+/**
+ * Multi user hosted rotation options
+ */
+export interface MultiUserHostedRotationOptions extends SingleUserHostedRotationOptions {
+  /**
+   * The master secret for a multi user rotation scheme
+   */
+  readonly masterSecret: ISecret;
+}
+
+/**
+ * Properties for a hosted rotation
+ */
+export interface HostedRotationProps extends SingleUserHostedRotationOptions {
+  /**
+   * The type of hosted rotation
+   */
+  readonly type: HostedRotationType;
+
+  /**
+   * The master secret for a multi user rotation scheme
+   *
+   * @default - single user rotation scheme
+   */
+  readonly masterSecret?: ISecret;
+}
+
+/**
+ * A hosted rotation
+ */
+export class HostedRotation implements ec2.IConnectable {
+  /**
+   * MySQL Single User
+   */
```

Comment:
`~~` (Super-nitpick): In this kinds of lengthy almost-copy-pasted blocks, I prefer to single-line docstrings where possible.
```suggestion
  /** MySQL Single User */
```